operator/Dockerfile

```diff
@@ -16,11 +16,11 @@ COPY controllers/ controllers/
 COPY internal/ internal/
 
 # Build
-RUN CGO_ENABLED=0 GOOS=linux GO111MODULE=on go build -mod=readonly -a -o manager main.go
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -mod=readonly -a -o manager main.go
 
 # Use distroless as minimal base image to package the manager binary
 # Refer to https://github.com/GoogleContainerTools/distroless for more details
-FROM gcr.io/distroless/static:nonroot
+FROM gcr.io/distroless/static-debian11:nonroot
```

Comment:
Why is this change needed? The `-debian11` images are only for debugging. However we use this dockerfile to build production images on [docker.io/grafana/loki-operator](https://hub.docker.com/r/grafana/loki-operator)

Reply:
Gotcha. I think then I just misunderstood the context behind the names described in the GitHub.